pkg/capacity/capacity_test.go

```diff
@@ -1427,3 +1528,317 @@ func makeSCs(in []testSC) (items []runtime.Object) {
 	}
 	return
 }
+
+func TestTermToSegment(t *testing.T) {
+	testcases := map[string]struct {
+		term          v1.NodeSelectorTerm
+		expectSegment topology.Segment
+		expectError   bool
+	}{
+		"matchfields": {
+			term: v1.NodeSelectorTerm{
+				MatchFields: []v1.NodeSelectorRequirement{
+					{
+						Key:    "name",
+						Values: []string{"worker-1"},
+					},
+				},
+			},
+			expectError: true,
+		},
+		"invalid-operator": {
+			term: v1.NodeSelectorTerm{
+				MatchExpressions: []v1.NodeSelectorRequirement{
+					{
+						Key:      "segment",
+						Operator: v1.NodeSelectorOpNotIn,
+						Values:   []string{"a"},
+					},
+				},
+			},
+			expectError: true,
+		},
+		"invalid-values": {
+			term: v1.NodeSelectorTerm{
+				MatchExpressions: []v1.NodeSelectorRequirement{
+					{
+						Key:      "segment",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"a", "b"},
+					},
+				},
+			},
+			expectError: true,
+		},
+		"simple": {
+			term: v1.NodeSelectorTerm{
+				MatchExpressions: []v1.NodeSelectorRequirement{
+					{
+						Key:      "segment",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"a"},
+					},
+				},
+			},
+			expectSegment: topology.Segment{topology.SegmentEntry{Key: "segment", Value: "a"}},
+		},
+		"multi": {
+			term: v1.NodeSelectorTerm{
+				MatchExpressions: []v1.NodeSelectorRequirement{
+					{
+						Key:      "segment",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"a"},
+					},
+					{
+						Key:      "zone",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"X"},
+					},
+				},
+			},
+			expectSegment: topology.Segment{
+				topology.SegmentEntry{Key: "segment", Value: "a"},
+				topology.SegmentEntry{Key: "zone", Value: "X"},
+			},
+		},
+		"unsorted": {
+			term: v1.NodeSelectorTerm{
+				MatchExpressions: []v1.NodeSelectorRequirement{
+					{
+						Key:      "zone",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"X"},
+					},
+					{
+						Key:      "segment",
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"a"},
+					},
+				},
+			},
+			expectSegment: topology.Segment{
+				topology.SegmentEntry{Key: "segment", Value: "a"},
+				topology.SegmentEntry{Key: "zone", Value: "X"},
+			},
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			segment, err := termToSegment(tc.term)
+			if tc.expectError && err == nil {
+				t.Fatalf("expected error, got segment %v", segment)
+			}
+			if !tc.expectError && err != nil {
+				t.Fatalf("expected no error, got: %v", err)
+			}
+			if segment.Compare(tc.expectSegment) != 0 {
+				t.Fatalf("expected segment %v, got %v", tc.expectSegment, segment)
+			}
+		})
+	}
+}
+
+func TestRefresh(t *testing.T) {
+	testcases := map[string]struct {
+		topology          *topology.Mock
+		initialSCs        []testSC
+		initialCapacities []testCapacity
```

Comment:
This is not set in test cases it looks like. This should be set in order for the controller's capacity cache to be populated and for refreshes to take action right?

Reply:
That's copy-and-pasted from `TestCapacity`. In `TestRefresh` it is not needed because `c.prepare` will create objects based on the initial topology and storage classes - that is what the tests then refresh. I removed the unused `initialCapacities`.